### DISCUSSION TITLE
Problem updating the rendering when overwriting a file

### DISCUSSION THREAD

**Discussion by odhondt** [developmentseed/titiler#816] [Q&A] [answered]
Titiler is a great tool but I was wondering: is there a way to clear the cache so that when a file is overwritten the preview is updated? 

I am using the default app in jupyter notebooks and even if I re-execute the cell, the image stays the same after a file is overwritten.
So far the only workaround I found is to add a small random value to the bounds of the `rescale` parameter (or to write the file with a different name every time, which is not really practical in my use case.)

**Comment by vincentsarago** [comment]
Hi @odhondt 

This has been previously discussed here but I can't find the ticket.

you can check https://github.com/rasterio/rasterio/issues/2181 and https://github.com/rasterio/rasterio/issues/2209

**Reply by odhondt** [reply]
Hi @vincentsarago 
Thanks.
So I understand it has something to do with GDAL VSI cache. 
What I still don't get is how to fix that using TiTiler.

**Reply by vincentsarago** [reply, marked as answer]
There is no direct `fix` in TiTiler itself. 

You could may do something like 


```python
from fastapi import Query
from rasterio._path import _parse_path as parse_path
from typing_extensions import Annotated


def EnvParams(url: Annotated[str, Query(description="Dataset URL")]) -> str:
    return {"CPL_VSIL_CURL_NON_CACHED": parse_path(url).as_vsi()}


from titiler.core.factory import TilerFactory
from fastapi import FastAPI

app = FastAPI()
cog = TilerFactory(environment_dependency=EnvParams)
app.include_router(cog.router)


```
ref: https://github.com/developmentseed/tilebench/blob/main/tilebench/viz.py#L140

You may also add `VSI_CACHE=False` and `GDAL_CACHEMAX=0` in your environment to force GDAL not too cache anything 

☝️ this will impact the performance. 

note: You may do something smarter by first check if the file has changed (e.g by checking the time of the file creation) before setting `CPL_VSIL_CURL_NON_CACHED` env.

**Reply by odhondt** [reply]
That's super interesting. So that means serving a custom app instead of using the default one, right?

**Reply by vincentsarago** [reply]
yes

**Reply by odhondt** [reply]
thanks
